Module 2

Deman4ig opened the pull request:
Привет!

Все готово!

Спасибо за проверку:)

Commit:
Add solution for module 2 (Streams)
1. task 1
2. task 3
3. task 4
4. task 5

Commit:
Add solution for module 2 (streams) task 2 (split stream)

jsru commented:
Добавляю преподавателя (@tadjik1) для код-ревью.

tadjik1 commented on 03-streams/01-limit-size-stream/LimitSizeStream.js:
```diff
@@ -4,10 +4,16 @@ const LimitExceededError = require('./LimitExceededError');
+      this._reminder -= chunk.byteLength;
+      callback(null, chunk);
+    } else {
+      this.destroy(new LimitExceededError());
```
Можно и так, но правильнее вызвать callback с ошибкой, и дестрой будет не нужен.

Deman4ig replied:
Тестировал оба варианта, но так и не понял разницы до конца:)

Переделал на callback()

tadjik1 commented on 03-streams/04-file-server-post/server.js:
```diff
@@ -1,17 +1,50 @@
+      const writeFile = util.promisify(fs.writeFile);
+      let currentSize = 0;
+
+      for await (const chunk of req) {
```
Интересный подход, но он рушит идею использования стримов, так как все данные тут собираются в память. Нужно переделать на использование метода .pipe() и использования стримов.

Deman4ig replied:
Дааа.. и этот подход будет фатальным если лимит 100ГБ:)

tadjik1 commented on 03-streams/05-file-server-delete/server.js:
```diff
@@ -1,17 +1,36 @@
 
   switch (req.method) {
     case 'DELETE':
+      stats(filepath).then(() => rm(filepath)).then(() => res.end()).catch((e) => {
```
А зачем тут stats? И без него все сработает, unlink тоже может зареджектится с таким кодом.

Deman4ig replied:
Согласен!

tadjik1 reviewed (commented):
@Deman4ig отличный оригинальный подход, и большинство задач и логики реализованы верно. Тем не менее есть пара моментов, над которыми стоит ещё поработать.

Commit:
Redo task04 with streams
Remove stats check in task05
Replace stream.destroy() with callback(err) in task01

jsru commented:
Добавляю преподавателя (@tadjik1) для код-ревью.

tadjik1 reviewed (approved):
👍 отлично
